walk/config.go

```diff
@@ -108,6 +98,10 @@ func (cr *Configurer) Configure(c *config.Config, rel string, f *rule.File) {
 				}
 				wcCopy.excludes = append(wcCopy.excludes, path.Join(rel, d.Value))
 			case "follow":
+				if err := checkPathMatchPattern(path.Join(rel, d.Value)); err != nil {
```

Comment:
Am I right to say that this has very minor potential for breaking existing uses, namely only if a file path that was previously interpreted literally contains characters with special meaning for `doublestar.Match`? That would seem acceptable, but we should probably mention the set of these characters and how to escape them in the docs for `# gazelle:follow`.

Reply:
I should note that prior to this change `follow` was set on `walkConfig` but never actually read or used, so once potentially source of breakage is extant uses of `follow` that now become active.